parser-typechecker/src/Unison/Codebase/Editor/Propagate.hs

```diff
@@ -465,11 +465,12 @@ applyPropagate patch Edits {..} = do
       (if isPropagated r'
        then Metadata.insert (propagatedMd (Referent.Ref r'))
        else Metadata.delete (propagatedMd (Referent.Ref r'))) .
-      Star3.replaceFact (Referent.Ref r) (Referent.Ref r') $ s
+      Star3.replaceFact (Referent.Ref r) (Referent.Ref r') .
+        Star3.mapD3 (\(tp, v) -> if v == r then (tp, r') else (tp, v)) $ s
```

Comment:
This is really all there is to this change. Update the range of the `d3` wherever the old reference occurs and point it to the new refefence instead.

Reply:
Is there any chance that `r` and `r'` would not have the same type `tp`?

Reply:
I suppose that's possible, yes. I'll need to look up the type of the new reference.

Reply:
And then in that situation, I guess you'll have to update D2 also :-\

Reply:
That's right. Fixed that now

Reply:
Nice catch. Code review ftw!